Feature/windows test

electronicpeter opened the pull request:
All changes done so far are for DOC-275 and DOC-272. The code can be compiled with mvn clean install -D defaultTests in windows and under macos. 
The .travis file has not been touched and has to be recovered to regular jobs.
In datasafe-business the code-analyze-task had to be deactivated because under windows it led to an unspecific error.

Commit:
new Strings always with CHARSET for UTF-8

electronicpeter committed:
fixed utf8 test

Commit:
replace backslash with forward slash in filecompares

Commit:
DOC-272 searching in ProfileRetrievalServiceImpl

Commit:
Dummy code to show deletion is possible.

Commit:
fix cleanup and stack for readDir

Commit:
remove Position class to avoid open streams of directory

Commit:
remove detailed logs

Commit:
added cleanup to find open filehandle

Commit:
DOC-275 open filehandles detected in testclass

Commit:
DOC-275 wrap reads in try resources

Commit:
DOC-275 wrap lists in try resources

Commit:
DOC-272 another unclosed list

Commit:
DOC-275 removed unecessary changes

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BaseE2ETest.java:
```diff
@@ -51,6 +52,7 @@
 @Slf4j
 @RequiredArgsConstructor
 public abstract class BaseE2ETest extends WithStorageProvider {
+    protected Charset THE_CHARSET=Charset.forName("UTF-8");
```
Instead of THE_CHARSET it is possible to import statically UTF_8 from StandardCharsets

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityTest.java:
```diff
@@ -225,7 +226,9 @@ private void validateInboxStructAndEncryption(AbsoluteLocation<ResolvedResource>
         assertThat(foundResource.location().getPath()).asString().contains(SHARED_FILE);
         // validate encryption on high-level:
-        assertThat(storage.read(foundResource)).asString().doesNotContain(MESSAGE_ONE);
+        try (java.io.InputStream read = storage.read(foundResource)) {
```
Remove fully qualified name of InputStream

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityTest.java:
```diff
@@ -240,15 +243,19 @@ private void validatePrivateStructAndEncryption(AbsoluteLocation<ResolvedResourc
         assertThat(foundResource.toString()).doesNotContain(PRIVATE_FILE);
         assertThat(foundResource.toString()).doesNotContain(FOLDER);
-        assertThat(storage.read(foundResource)).asString().doesNotContain(MESSAGE_ONE);
+        try (java.io.InputStream read = storage.read(foundResource)) {
```
Remove fully qualified name of InputStream

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityTest.java:
```diff
@@ -240,15 +243,19 @@ private void validatePrivateStructAndEncryption(AbsoluteLocation<ResolvedResourc
-                .filter(it -> !it.location().toASCIIString().startsWith("."))
-                .filter(it -> pattern.test(it.location().toASCIIString()))
-                .collect(Collectors.toList());
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = storage.list(new AbsoluteLocation<>(BasePrivateResource.forPrivate(location)))) {
```
Maybe `ls` instead of `stream` is short and understandable for unix users variable name

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/DataTamperingResistanceTest.java:
```diff
@@ -105,8 +106,11 @@ void testPrivateDocumentPathTamperResistance(String path) {
-                () -> listPrivate.list(ListRequest.forDefaultPrivate(jane, ""))
-                        .forEach(it -> log.info("{}", it.location())) // consume stream
+                () -> {
+                    try (Stream<AbsoluteLocation<ResolvedResource>> stream = listPrivate.list(ListRequest.forDefaultPrivate(jane, ""))) {
```
Can we have more meaningful name for `stream`

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/SchemeDelegationWithDbTest.java:
```diff
@@ -101,13 +103,15 @@ void testProfileOnDbDataOnFsWorks() {
     private Stream<String> listDb(String path) {
-        return db.list(BasePrivateResource.forAbsolutePrivate(URI.create(path)))
-            .map(it -> it.location().asURI().toString());
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = db.list(BasePrivateResource.forAbsolutePrivate(URI.create(path)))){
```
Try-with-resources is meaningless here - opened stream is manipulated elsewhere, better to return List from this method and keep try-with-resources

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/SchemeDelegationWithDbTest.java:
```diff
@@ -91,6 +92,7 @@ void testProfileOnDbDataOnFsWorks() {
         assertThat(Files.walk(fsPath))
             .extracting(it -> fsPath.relativize(it))
             .extracting(Path::toString)
+                .extracting(string -> string.replaceAll("\\\\","/"))
```
We expect relative paths here, so Path.toURI().getPath could work better

valb3r commented on datasafe-directory/datasafe-directory-impl/src/main/java/de/adorsys/datasafe/directory/impl/profile/operations/actions/ProfileRemovalServiceImpl.java:
```diff
@@ -115,8 +116,8 @@ private void removeAllIn(UserIDAuth userID, AbsoluteLocation<PrivateResource> lo
-        listService.list(
-                access.privateAccessFor(userID, location.getResource())
-        ).forEach(removeService::remove);
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = listService.list(access.privateAccessFor(userID, location.getResource()))) {
```
Could we have more meaningful name here for stream?

valb3r commented on datasafe-simple-adapter/datasafe-simple-adapter-impl/src/main/java/de/adorsys/datasafe/simple/adapter/impl/SimpleDatasafeServiceImpl.java:
```diff
@@ -251,9 +249,11 @@ public void deleteFolder(UserIDAuth userIDAuth, DocumentDirectoryFQN documentDir
     @Override
     public void cleanupDb() {
-        storageService
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = storageService
```
Could we have more meaningful name here for stream?

valb3r commented on datasafe-simple-adapter/datasafe-simple-adapter-impl/src/test/java/de/adorsys/datasafe/simple/adapter/impl/CleanupDbTest.java:
```diff
@@ -46,18 +50,14 @@ void cleanupDb(WithStorageProvider.StorageDescriptor descriptor) {
-                        BasePrivateResource.forAbsolutePrivate(descriptor.getLocation())
-                ).count()
-        );
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = descriptor.getStorageService().get().list(BasePrivateResource.forAbsolutePrivate(descriptor.getLocation()))) {
```
Could we have more meaningful name here for stream?

valb3r commented on datasafe-simple-adapter/datasafe-simple-adapter-impl/src/test/java/de/adorsys/datasafe/simple/adapter/impl/CleanupDbTest.java:
```diff
@@ -46,18 +50,14 @@ void cleanupDb(WithStorageProvider.StorageDescriptor descriptor) {
-                        BasePrivateResource.forAbsolutePrivate(descriptor.getLocation())
-                ).count()
-        );
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = descriptor.getStorageService().get().list(BasePrivateResource.forAbsolutePrivate(descriptor.getLocation()))) {
```
Could we have more meaningful name here for stream?

valb3r commented on datasafe-simple-adapter/datasafe-simple-adapter-impl/src/test/java/de/adorsys/datasafe/simple/adapter/impl/DFSRelativeToRootProfileTest.java:
```diff
@@ -53,8 +56,8 @@ void createWriteAndDeleteUserCleansAllNeeded(WithStorageProvider.StorageDescript
-        assertThat(descriptor.getStorageService().get().list(
-                BasePrivateResource.forAbsolutePrivate(descriptor.getLocation()))
-        ).isEmpty();
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = descriptor.getStorageService().get().list(BasePrivateResource.forAbsolutePrivate(descriptor.getLocation()))) {
```
Could we have more meaningful name here for stream?

valb3r commented on datasafe-simple-adapter/datasafe-simple-adapter-impl/src/test/java/de/adorsys/datasafe/simple/adapter/impl/SimpleAdapterFeatureTest.java:
```diff
@@ -53,6 +55,7 @@ void testWithEncryption() {
         AbsoluteLocation<PrivateResource> rootLocation = getPrivateResourceAbsoluteLocation();
         Assertions.assertEquals(0, simpleDatasafeService.getStorageService().list(rootLocation).filter(el -> el.location().toASCIIString().contains(path)).count());
         simpleDatasafeService.destroyUser(userIDAuth);
+        simpleDatasafeService.cleanupDb();
```
Why this is needed here?

valb3r commented on datasafe-simple-adapter/datasafe-simple-adapter-impl/src/test/java/de/adorsys/datasafe/simple/adapter/impl/SimpleAdapterFeatureTest.java:
```diff
@@ -64,13 +67,20 @@ void testWithoutPathEncryption() {
+            Assertions.assertEquals(1, absoluteLocationStream.count());
+        }
+        try (Stream<AbsoluteLocation<ResolvedResource>> absoluteLocationStream = simpleDatasafeService.getStorageService().list(rootLocation).filter(el -> el.location().toASCIIString().contains(path))) {
+            Optional<AbsoluteLocation<ResolvedResource>> first =absoluteLocationStream.findFirst();
```
Needs spaces - ` = `

valb3r commented on datasafe-simple-adapter/datasafe-simple-adapter-impl/src/test/java/de/adorsys/datasafe/simple/adapter/impl/SimpleAdapterFeatureTest.java:
```diff
@@ -64,13 +67,20 @@ void testWithoutPathEncryption() {
+            }
+        }
         simpleDatasafeService.destroyUser(userIDAuth);
+        simpleDatasafeService.cleanupDb();
```
If this is needed - it is better to move it to AfterEach

valb3r commented on datasafe-simple-adapter/datasafe-simple-adapter-impl/src/test/java/de/adorsys/datasafe/simple/adapter/impl/SimpleAdapterFeatureTest.java:
```diff
@@ -84,13 +94,20 @@ void testWithoutEncryption() {
+            }
+        }
         simpleDatasafeService.destroyUser(userIDAuth);
+        simpleDatasafeService.cleanupDb();
```
Maybe move this to AfterEach?

valb3r commented on datasafe-storage/datasafe-storage-api/src/test/java/de/adorsys/datasafe/storage/api/SchemeDelegatingStorageTest.java:
```diff
@@ -46,16 +50,22 @@ void objectExistsDelegates() {
     @Test
     void listDelegates() {
-        tested.list(locationExists);
+        try (Stream<AbsoluteLocation<ResolvedResource>> list = tested.list(locationExists)) {
```
Not necessary here since backing service is a mock

valb3r commented on datasafe-storage/datasafe-storage-api/src/test/java/de/adorsys/datasafe/storage/api/SchemeDelegatingStorageTest.java:
```diff
@@ -46,16 +50,22 @@ void objectExistsDelegates() {
-        verify(service).list(locationExists);
+        }
+
+        try (Stream<AbsoluteLocation<ResolvedResource>> list = verify(service).list(locationExists)) {
```
Not necessary here since backing service is a mock

valb3r commented on datasafe-storage/datasafe-storage-api/src/test/java/de/adorsys/datasafe/storage/api/SchemeDelegatingStorageTest.java:
```diff
@@ -46,16 +50,22 @@ void objectExistsDelegates() {
-        tested.read(locationExists);
-
-        verify(service).read(locationExists);
+        try (InputStream read = tested.read(locationExists)) {
```
Not necessary here since backing service is a mock

valb3r commented on datasafe-storage/datasafe-storage-api/src/test/java/de/adorsys/datasafe/storage/api/UriBasedAuthStorageServiceTest.java:
```diff
@@ -54,17 +57,22 @@ void objectExists(MappedItem item) {
     @ParameterizedTest
     void list(MappedItem item) {
-        tested.list(item.getUri());
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = tested.list(item.getUri())) {
```
Not necessary here - is a mock

valb3r commented on datasafe-storage/datasafe-storage-api/src/test/java/de/adorsys/datasafe/storage/api/UriBasedAuthStorageServiceTest.java:
```diff
@@ -54,17 +57,22 @@ void objectExists(MappedItem item) {
-        tested.read(item.getUri());
-        assertThat(argumentCaptor.getValue()).isEqualToComparingFieldByField(item.getAccessId());
-        verify(storage).read(item.getUri());
+        try (InputStream read = tested.read(item.getUri())) {
```
Not necessary here - is a mock

valb3r commented on datasafe-storage/datasafe-storage-impl-db/src/test/java/de/adorsys/datasafe/storage/impl/db/DatabaseStorageServiceTest.java:
```diff
@@ -71,21 +73,28 @@ void objectExists() {
-        assertThat(storageService.list(FILE))
-                .extracting(it -> it.getResource().location().asString())
-                .containsOnly("jdbc://localhost:9999/h2:mem:test/private_profiles/path/hello.txt");
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = storageService.list(FILE)) {
```
Maybe we could have more meaningful name for `stream` ?

valb3r commented on datasafe-storage/datasafe-storage-impl-db/src/test/java/de/adorsys/datasafe/storage/impl/db/DatabaseStorageServiceTest.java:
```diff
@@ -71,21 +73,28 @@ void objectExists() {
-                        "jdbc://localhost:9999/h2:mem:test/private_profiles/path/hello.txt",
-                        "jdbc://localhost:9999/h2:mem:test/private_profiles/path/hello1.txt"
-                );
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = storageService.list(ROOT)) {
```
Maybe we could have more meaningful name for `stream` ?

valb3r commented on datasafe-storage/datasafe-storage-impl-fs/src/test/java/de/adorsys/datasafe/storage/impl/fs/FileSystemStorageServiceTest.java:
```diff
@@ -45,14 +46,18 @@ void prepare(@TempDir Path dir) {
     void objectExists() {
         createFileWithMessage();
-        assertThat(storageService.objectExists(storageService.list(root).findFirst().get())).isTrue();
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = storageService.list(root)) {
```
Maybe we could have more meaningful name for `stream` ?

valb3r commented on datasafe-storage/datasafe-storage-impl-fs/src/test/java/de/adorsys/datasafe/storage/impl/fs/FileSystemStorageServiceTest.java:
```diff
@@ -45,14 +46,18 @@ void prepare(@TempDir Path dir) {
         Path nonExistingFile = storageDir.resolve(UUID.randomUUID().toString());
         AbsoluteLocation<PrivateResource> nonExistingFileLocation = new AbsoluteLocation<>(BasePrivateResource.forPrivate(nonExistingFile.toUri()));
-        assertThat(storageService.list(nonExistingFileLocation).collect(Collectors.toList())).isEmpty();
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = storageService.list(nonExistingFileLocation)) {
```
Maybe we could have more meaningful name for `stream` ?

valb3r commented on datasafe-storage/datasafe-storage-impl-fs/src/test/java/de/adorsys/datasafe/storage/impl/fs/FileSystemStorageServiceTest.java:
```diff
@@ -45,14 +46,18 @@ void prepare(@TempDir Path dir) {
         AbsoluteLocation<PrivateResource> nonExistingFileLocation = new AbsoluteLocation<>(BasePrivateResource.forPrivate(nonExistingFile.toUri()));
-        assertThat(storageService.list(nonExistingFileLocation).collect(Collectors.toList())).isEmpty();
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = storageService.list(nonExistingFileLocation)) {
+            assertThat(stream.collect(Collectors.toList())).isEmpty();
```
Can be
 assertThat(stream).isEmpty();

valb3r commented on datasafe-storage/datasafe-storage-impl-fs/src/test/java/de/adorsys/datasafe/storage/impl/fs/FileSystemStorageServiceTest.java:
```diff
@@ -104,33 +109,43 @@ void listDotFilesToo() {
 
-        assertThat(storageService.list(root).collect(Collectors.toList())).isEmpty();
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = storageService.list(root)) {
+            assertThat(stream.collect(Collectors.toList())).isEmpty();
```
Can be
assertThat(stream).isEmpty();

valb3r commented on datasafe-storage/datasafe-storage-impl-fs/src/test/java/de/adorsys/datasafe/storage/impl/fs/FileSystemStorageServiceTest.java:
```diff
@@ -104,33 +109,43 @@ void listDotFilesToo() {
         }
-        assertThat(storageService.list(root).collect(Collectors.toList())).isNotEmpty();
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = storageService.list(root)) {
+            assertThat(stream.collect(Collectors.toList())).isNotEmpty();
```
Can be
assertThat(stream).isNotEmpty();

valb3r commented on datasafe-storage/datasafe-storage-impl-fs/src/test/java/de/adorsys/datasafe/storage/impl/fs/FileSystemStorageServiceTest.java:
```diff
@@ -104,33 +109,43 @@ void listDotFilesToo() {
-                .hasSize(1)
-                .extracting(AbsoluteLocation::location)
-                .asString().contains(FILE);
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = storageService.list(root)) {
```
Maybe we could have more meaningful name for stream ?

valb3r commented on datasafe-storage/datasafe-storage-impl-fs/src/test/java/de/adorsys/datasafe/storage/impl/fs/FileSystemStorageServiceTest.java:
```diff
@@ -104,33 +109,43 @@ void listDotFilesToo() {
     @Test
     void listOnNonExisting() {
-        assertThat(storageService.list(root)).isEmpty();
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = storageService.list(root)) {
```
Maybe we could have more meaningful name for stream ?

Commit:
DOC-272 make docker available even with other host

Commit:
DOC-275 close all touched files

Commit:
DOC-275 mvn install under windows leads to unspecified error, so analyze fail does not stop build any more.

[INFO] datasafe-storage-impl-db ........................... SUCCESS [ 19.781 s]
[INFO] datasafe-business .................................. FAILURE [04:47 min]
[INFO] datasafe-rest-impl ................................. SKIPPED
[INFO] datasafe-examples .................................. SKIPPED
[INFO] datasafe-examples-business ......................... SKIPPED
[INFO] datasafe-examples-customize-dagger ................. SKIPPED
[INFO] datasafe-examples-versioned-s3 ..................... SKIPPED
[INFO] datasafe-examples-multidfs ......................... SKIPPED
[INFO] datasafe-simple-adapter ............................ SKIPPED
[INFO] datasafe-simple-adapter-api ........................ SKIPPED
[INFO] datasafe-simple-adapter-impl ....................... SKIPPED
[INFO] datasafe-simple-adapter-spring ..................... SKIPPED
[INFO] datasafe-long-run-tests ............................ SKIPPED
[INFO] datasafe-business-tests-random-actions ............. SKIPPED
[INFO] last-module-codecoverage-check ..................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  11:32 min
[INFO] Finished at: 2019-10-02T11:19:44+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.1.1:analyze-only (analyze) on project datasafe-business: Dependency problems found -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :datasafe-business

y:\DATASAFE\datasafe>mvn -version
Apache Maven 3.6.2 (40f52333136460af0dc0d7232c0dc0bcf0d9e117; 2019-08-27T17:06:16+02:00)
Maven home: c:\opt\apache-maven-3.6.2\bin\..
Java version: 1.8.0_221, vendor: Oracle Corporation, runtime: c:\opt\java8\jre
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

Commit:
DOC-275
BC needs to be loaded so that test can run on windows too

Commit:
DOC-275 first time compilable under windows

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityWithConcurrencyTest.java:
```diff
@@ -109,6 +110,7 @@ void writeToPrivateListPrivateInDifferentThreads(WithStorageProvider.StorageDesc
         CountDownLatch finishHoldingLatch = new CountDownLatch(NUMBER_OF_TEST_USERS * NUMBER_OF_TEST_FILES);
 
         String checksumOfOriginTestFile;
+        log.trace("*** get checksum of "+testFile+" ***");
```
Use string interpolation for logging

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityWithConcurrencyTest.java:
```diff
@@ -153,11 +155,15 @@ void writeToPrivateListPrivateInDifferentThreads(WithStorageProvider.StorageDesc
         metricCollector.setNumberOfThreads(poolSize);
         metricCollector.writeToJSON();//json files in target folder
+
+        deleteTestFile(testFile);
```
Should be handled by afterEach or by TempDir directive

electronicpeter replied:
why. Test creates with generateTestFile and eventually deletes with deleteTestfile. Otherwise generateTestFile should be done in BeforeEach too. But I did not want to change too much.

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityWithConcurrencyTest.java:
```diff
@@ -153,11 +155,15 @@ void writeToPrivateListPrivateInDifferentThreads(WithStorageProvider.StorageDesc
     private List<AbsoluteLocation<ResolvedResource>> listAllPrivateFiles(UserIDAuth user) {
-        return listPrivate.list(
-                forDefaultPrivate(user, "./")).collect(Collectors.toList());
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = listPrivate.list(forDefaultPrivate(user, "./"))) {
```
More meaningful name, not stream please - ls is okay

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityWithConcurrencyTest.java:
```diff
@@ -188,6 +194,7 @@ private void createFileForUserParallelly(ThreadPoolExecutor executor, CountDownL
 
                     log.debug("Save file in {} ms", durationOfSavingFile);
                 } catch (InterruptedException e) {
+                    Arrays.stream(e.getStackTrace()).forEach(el -> log.error(el.toString()));
```
We don't really need this

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityWithConcurrencyTest.java:
```diff
@@ -197,12 +204,13 @@ private void createFileForUserParallelly(ThreadPoolExecutor executor, CountDownL
+                return checksumOfDecryptedTestFile;
+            }
         } catch (IOException e) {
+            Arrays.stream(e.getStackTrace()).forEach(el -> log.error(el.toString()));
```
We don't really need this

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityWithConcurrencyTest.java:
```diff
@@ -221,6 +229,16 @@ private String checksum(InputStream input) {
+        // and following tests were able to reuse directory. Now file handles are closed
+        // and all files including directory are deleted.
+        if (!directory.exists()) {
+            log.trace(directory + " does not exist. will be created now");
```
Use {} string interpolation

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityWithConcurrencyTest.java:
```diff
@@ -221,6 +229,16 @@ private String checksum(InputStream input) {
     }
 
     private static void generateTestFile(String testFile, int testFileSizeInBytes) {
+        log.trace("*** generate "+testFile+" ***");
```
Use {} string interpolation

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityWithConcurrencyTest.java:
```diff
@@ -229,11 +247,26 @@ private static void generateTestFile(String testFile, int testFileSizeInBytes) {
             }
         } catch (IOException e) {
-            log.error(e.getMessage());
+            log.error("generateTestFile: " + e.getMessage());
```
Use string interpolation

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityWithConcurrencyTest.java:
```diff
@@ -272,15 +305,13 @@ private void init(WithStorageProvider.StorageDescriptor descriptor) {
-
-            closeQuietly(fis);
-            closeQuietly(write);
+            try (OutputStream write = writeToPrivate.write(WriteRequest.forDefaultPrivate(john, filePathForWriting))) {
```
Nested try statement is not necessary, try-with resources supports nesting try (is =...; fs =...) will work fine

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/DataTamperingResistanceTest.java:
```diff
@@ -95,7 +97,9 @@ void testInboxDocumentContentTamperResistance() {
     @SneakyThrows
-    void testPrivateDocumentPathTamperResistance(String path) {
+    void testPrivateDocumentPathTamperResistance(String origpath) {
+        String path = origpath.replaceAll("/", File.pathSeparator);
```
This shouldn't be needed - it is re-encoded into URI canonical form

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/MultiDFSFunctionalityTest.java:
```diff
@@ -338,4 +339,14 @@ private WithCredentialProvider(Lazy<StorageKeyStoreOperations> storageKeyStoreOp
+            return uri;
+        }
+        URI dockeruri = new URI(dockerhost);
+        return "http://" + dockeruri.getHost();
```
It is better to return host url, for usecase when we need https for example

valb3r commented on datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/SchemeDelegationWithDbTest.java:
```diff
@@ -101,13 +103,15 @@ void testProfileOnDbDataOnFsWorks() {
                 "users/john/private/files",
                 "users/john/private/files/SIV",
-                fsPath.relativize(encryptedFile).toString()
+                fsPath.relativize(encryptedFile).toString().replaceAll("\\\\","/")
```
We expect relative paths here, so Path.toURI().getPath could work better

valb3r commented on datasafe-business/src/test/resources/simplelogger.properties:
```diff
@@ -1,7 +1,13 @@
 org.slf4j.simpleLogger.logFile=System.out
+org.slf4j.simpleLogger.levelInBrackets=true
+
+# org.slf4j.simpleLogger.log.de.adorsys.datasafe.teststorage.WithStorageProvider=debug
```
Remove commented code please

valb3r commented on datasafe-examples/datasafe-examples-multidfs/src/test/java/de/adorsys/datasafe/examples/business/s3/MultiDfsWithCredentialsExampleTest.java:
```diff
@@ -66,12 +67,16 @@
     @BeforeAll
     static void startup() {
+        // on windows this is required
+        log.debug("Add Bouncy Castle to test");
```
Please drop unnecessary log statement

valb3r commented on datasafe-examples/datasafe-examples-multidfs/src/test/java/de/adorsys/datasafe/examples/business/s3/MultiDfsWithCredentialsExampleTest.java:
```diff
@@ -237,4 +242,14 @@ private WithCredentialProvider(Lazy<StorageKeyStoreOperations> storageKeyStoreOp
+            return uri;
+        }
+        URI dockeruri = new URI(dockerhost);
+        return "http://" + dockeruri.getHost();
```
Better to set docker uri in environment variable

valb3r commented on datasafe-examples/datasafe-examples-multidfs/src/test/resources/simplelogger.properties:
```diff
@@ -5,3 +5,5 @@ org.slf4j.simpleLogger.showDateTime=true
 
 org.slf4j.simpleLogger.logFile=System.out
+
+# org.slf4j.simpleLogger.log.de.adorsys.datasafe.examples.business.s3.MultiDfsWithCredentialsExampleTest=debug
```
Drop commented code please

valb3r commented on datasafe-simple-adapter/datasafe-simple-adapter-impl/src/test/java/de/adorsys/datasafe/simple/adapter/impl/SimpleDatasafeAdapter043CompatTest.java:
```diff
@@ -67,8 +67,10 @@ void writeNewAndReadFileFromOldVersion() {
         // validate folder structure
         assertThat(Dirs.walk(dfsRoot, 1)).containsExactlyInAnyOrder("profiles", "users");
         assertThat(Dirs.walk(dfsRoot.resolve("profiles")))
+                .extracting(string -> string.replaceAll("\\\\","/"))
```
Better to use URI form

electronicpeter replied:
How would that look like?

valb3r replied:
Actually replaceAll seem to be useless here

valb3r commented on datasafe-simple-adapter/datasafe-simple-adapter-impl/src/test/java/de/adorsys/datasafe/simple/adapter/impl/SimpleDatasafeAdapterTest.java:
```diff
@@ -67,6 +69,25 @@ void myafter() {
+        // SimpleDatasafeAdapter does not use user profile json files, so only keystore and pubkeys should exist:
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = descriptor.getStorageService().get().list(BasePrivateResource.forAbsolutePrivate(descriptor.getLocation()))) {
+            assertThat(stream).extracting(it -> descriptor.getLocation().relativize(it.location()).asString())
+                    .extracting(string -> string.replaceAll("\\\\","/"))
```
What is the purpose of replacing \\ with / after relativization?

electronicpeter replied:
Hm, seems not really helpfull. yes.

valb3r commented on datasafe-simple-adapter/datasafe-simple-adapter-impl/src/test/java/de/adorsys/datasafe/simple/adapter/impl/SimpleDatasafeAdapterTest.java:
```diff
@@ -75,14 +96,14 @@ void justCreateAndDeleteUser(WithStorageProvider.StorageDescriptor descriptor) {
-
+        try (Stream<AbsoluteLocation<ResolvedResource>> stream = descriptor.getStorageService().get().list(BasePrivateResource.forAbsolutePrivate(descriptor.getLocation()))) {
+            assertThat(stream).extracting(it -> descriptor.getLocation().relativize(it.location()).asString())
+                    .extracting(string -> string.replaceAll("\\\\","/"))
```
What is the purpose of replacing \ with / after relativization?
Expectation does not expect '//'

electronicpeter replied:
Hm, really strange. Propably did it in advance. Yes. Unnecessary code.

valb3r commented on datasafe-test-storages/src/main/java/de/adorsys/datasafe/teststorage/WithStorageProvider.java:
```diff
@@ -404,4 +453,15 @@ public String getMappedUrl() {
+            return uri;
+        }
+        URI dockeruri = new URI(dockerhost);
+        return "http://" + dockeruri.getHost();
```
Better to have docker host uri and not hostname

electronicpeter replied:
I do not understand.

valb3r replied:
Docker host might be proxied too - `return "http://" + dockeruri;` should be better

valb3r reviewed (changes requested):
Please use string interpolation for logging

Commit:
DOC-272 review requests

Commit:
DOC-272 review requests

valb3r committed:
Dropped unused dependecy

valb3r committed:
Prettification of stuff
